force_wfmanager/wfmanager_setup_task.py

```diff
@@ -505,7 +458,142 @@ def prepare_destroy(self):
+            result = confirm(
+                None,
+                "Are you sure you want to run the computation and "
+                "empty the result table?",
+            )
+            if result is not YES:
+                return
+
+        # Run any plugin injected ui hooks before execution
+        # For example, the UI Notification Hooks Manager sets up sockets
+        # to communicate with the server before executing a workflow
+        try:
+            for hook_manager in self.ui_hooks_managers:
+                try:
+                    hook_manager.before_execution(self)
+                except Exception:
+                    log.exception(
+                        "Failed before_execution hook "
+                        "for hook manager {}".format(
+                            hook_manager.__class__.__name__
+                        )
+                    )
+
+            # Creates a temporary file containing the workflow
+            tmpfile_path = tempfile.mktemp()
+            write_workflow_file(self.workflow_model, tmpfile_path)
+
+            # Clear the analysis model before attempting to run
+            self.analysis_model.clear()
+
+            # Execute the bdss on a different thread
+            future = self.executor.submit(self._execute_bdss, tmpfile_path)
+            future.add_done_callback(self._execution_done_callback)
+        except Exception as e:
+            logging.exception("Unable to run BDSS.")
+            error(
+                None,
+                "Unable to run BDSS: {}".format(e),
+                "Error when running BDSS",
+            )
+            self.computation_running = False
+
+    def stop_bdss(self):
+        self.zmq_server.publish_message("STOP_BDSS")
+        self._paused = False
+
+    def pause_bdss(self):
+        """Pause an MCO run"""
+        if self._paused:
+            message = "RESUME_BDSS"
+        else:
+            message = "PAUSE_BDSS"
+        self.zmq_server.publish_message(message)
+        self._paused = not self._paused
+
+    # Plugin Status
+    def lookup_plugins(self):
+
+        plugins = [
+            plugin
+            for plugin in self.window.application.plugin_manager
+            if isinstance(plugin, BaseExtensionPlugin)
+        ]
+
+        # Plugins guaranteed to have an id, so sort by that if name is not set
+        plugins.sort(
+            key=lambda s: s.name if s.name not in ("", None) else s.id
+        )
+
+        return plugins
+
+    def open_plugins(self):
+        """Opens a dialogue window displaying information about the currently
+        loaded plugins
+        """
+        plugins = self.lookup_plugins()
+
+        dlg = PluginDialog(plugins)
+        dlg.edit_traits()
+
+    # Menu/Toolbar Methods
+    def switch_task(self):
+        """Switches to the review task and verifies startup setting are
+        correct for toolbars/menus etc."""
+        self.window.activate_task(self.review_task)
+
+    def exit(self):
```

Comment:
I realize that this is copied code, but `exit` is an application-level thing rather than a Task-level thing, conceptually.

The default exit behaviour for Tasks applications is here: https://github.com/enthought/envisage/blob/master/envisage/ui/tasks/tasks_application.py#L273-L298

This should do exactly what you want, so it is this method that you want to call, rather than this one.  There is a `TasksApplicationAction` that is suited for this use.

Reply:
Couldn't find much doc on `ApplicationTask`, so couldn't figure out how to implement that. But `set_task.exit` now calls the application's exit method.

Reply:
I think we should deal with this in a new issue / PR, since it lies a bit out of scope of the current work